torch/_inductor/codegen/cpp.py

```diff
@@ -3216,6 +3162,146 @@ def select_tiling(
+                )
+                op_counter: Dict[str, int] = {}
+                # ops may not cause overhead with vectorization, like non-contiguous
+                # index_expr, load, store
+                non_contig_indexing_op_counter: Dict[str, int] = {}
+
+                for _body in loop_bodies:
+                    sub_blocks = [_body.root_block] + list(_body.subblocks.values())
+                    for sub_block in sub_blocks:
+                        for _node in sub_block.graph.nodes:
+                            if _node.target == "index_expr":
+                                # Get the index and replace prefix from z to x
+                                index = sub_block.body.indexing_from_args(
+                                    (vars, reduction_vars)
+                                )[_node.args[-2].args[0]]
+                                if _is_valid_indices(itervars, tiling_indices):
+                                    stride = _try_get_stride(
+                                        index, itervars, tiling_factor, tiling_indices
+                                    )
+                                    if not (stride == 0 or stride is not None):
+                                        _update_negative_op_count(
+                                            _node.target, non_contig_indexing_op_counter
+                                        )
+                            elif _node.target == "load":
+                                index = sub_block.body.indexing_from_args(
+                                    (vars, reduction_vars)
+                                )[_node.args[-1].args[0]]
+                                if _is_valid_indices(itervars, tiling_indices):
+                                    stride = _try_get_stride(
+                                        index, itervars, tiling_factor, tiling_indices
+                                    )
+                                    if stride not in [0, 1]:
+                                        _update_negative_op_count(
+                                            _node.target, non_contig_indexing_op_counter
+                                        )
+                            elif _node.target == "store":
+                                index = sub_block.body.indexing_from_args(
+                                    (vars, reduction_vars)
+                                )[_node.args[2].args[0]]
+                                if _is_valid_indices(itervars, tiling_indices):
+                                    stride = _try_get_stride(
+                                        index, itervars, tiling_factor, tiling_indices
+                                    )
+                                    if stride != 1:
+                                        _update_negative_op_count(
+                                            _node.target, non_contig_indexing_op_counter
+                                        )
```

Comment:
Perhaps we can combine these if-elif code blocks to simplify the code?

Reply:
Sure, merge the code block for `index_expr`, `load` and `store`. Would like explain more about why different contiguous check function of stride used for these 3 ops which are trying to align the logic in `vec codegen` as https://github.com/pytorch/pytorch/blob/9f17037e8ba7d98d8fbbcf40625fcad48c4c0b16/torch/_inductor/codegen/cpp.py#L1459-L1461